3_extracting_features/DPutils.py

```diff
@@ -0,0 +1,220 @@
+from importlib.resources import path
+import pandas as pd
+import pathlib
+import shutil
+
+from PIL import Image
+import os
+
+def copy_DP_files(
+    project_path: pathlib.Path,
+    config_name: str,
+    checkpoint_name: str,
+):
+    """Copy config and checkpoint files to their designated location in DP project as assigned by the project_path
+    Args:
+        project_path (pathlib.Path): Path for DP project to be located in
+        config_name (str): Name of config file to copy
+        checkpoint_name (str): Name of checkpoint file to copy
+    """
+
+    # Copy config file to DP project
+    config_load_path = pathlib.Path(f"DP_files/{config_name}")
+    config_save_path = pathlib.Path(f"{project_path}/inputs/config/{config_name}")
+    config_save_path.parents[0].mkdir(parents=True, exist_ok=True)
+    shutil.copyfile(config_load_path, config_save_path)
+
+    # Copy checkpoint file to DP project
+    checkpoint_load_path = pathlib.Path(f"DP_files/{checkpoint_name}")
+    checkpoint_save_path = pathlib.Path(
+        f"{project_path}/outputs/efn_pretrained/checkpoint/{checkpoint_name}"
+    )
+    checkpoint_save_path.parents[0].mkdir(parents=True)
+    shutil.copyfile(checkpoint_load_path, checkpoint_save_path)
+
+def compile_index_csv(
+    images_load_path: pathlib.Path,
+    DP_images_path: pathlib.Path,
+    annotations: pd.DataFrame,
+    object: str,
+) -> pd.DataFrame:
+    """Compiles index csv file (image metadata, channel image locations, genotype)
+    Args:
+        images_load_path (pathlib.Path): Path to load illuminated corrected images from 
+        DP_images_path (pathlib.Path): Path to DP project images folder (DP_project/inputs/images)
+        annotations (pd.DataFrame): NF1 data annotations metadata csv file
+        object (str): Object to compile index csv for, either "nuc" (nucleus) or "cyto" (cytoplasm)
+    Returns:
+        pd.DataFrame: index csv dataframe
+    """
+    # Empty data frame for index to be appended to
+    index_csv_data = []
+
+    for image_paths in images_load_path.iterdir():
+        # Skip over files without DAPI in image path name
+        if "DAPI" not in image_paths.name:
+            continue
+
+        # Get image metadata
+        plate = int(image_paths.name[3:5])
+        well = image_paths.name[0:2]
+        site = image_paths.name[8]
+
+        # Get genotype value for images, assign plate and well columns from annotations for index
+        image_annotations = annotations.loc[
+            (plate == annotations["Plate"]) & (annotations["Well"] == well)
+        ]
+        genotype = image_annotations.iloc[0]["Genotype"]
+            
+        # Compile object index file data
+        # Note: In the NF1 data, the "RNA" channel is actually the "Actin" channel, but wanted to follow the same naming as LUAD paper
```

Comment:
Does it break the code to use the correct name? I worry that being inaccurate here will propagate confusion.

Reply:
I can try to change the name in the config file and and see what happens.

Reply:
@gwaybio I changed the "RNA" naming to "Actin" and DeepProfiler ran smoothly. Looks like this naming convention doesn't need to specific or stay the name as the LUAD project. It just needs to be the same in the `index.csv` and in the `config file (.json)`.

Reply:
Wonderful! Let's use actin then (please update the PR if you haven't already)